zebedee/client.go

```diff
@@ -0,0 +1,302 @@
+
+	for _, element := range related {
+		for i, e := range element {
+			sem <- 1
+			wg.Add(1)
+			go func(i int, e model.Related, element []model.Related) {
+				defer func() {
+					<-sem
+					wg.Done()
+				}()
+				element[i].Title = c.getPageTitle(e.URI)
+			}(i, e, element)
+		}
+	}
+	wg.Wait()
+
+	return c.populateStaticLandingPageModel(*dlp)
+}
+
+func (c ZebedeeClient) SetAccessToken(token string) {
+	log.Debug("adding access token to client", log.Data{"token": token})
+	c.accessToken = token
+}
+
+func (c ZebedeeClient) get(path string) ([]byte, error) {
+	req, err := http.NewRequest("GET", c.zebedeeURL+path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(c.accessToken) > 0 {
+		req.Header.Set("X-Florence-Token", c.accessToken)
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, ErrInvalidZebedeeResponse{errors.New("unexpected response from zebedee"), zebedeeRequestErrorData{
+			"zebedeeURI":         req.URL.Path,
+			"expectedStatusCode": 200,
+			"actualStatusCode":   resp.StatusCode,
+		}}
+	}
+
+	return ioutil.ReadAll(resp.Body)
+}
+
+func (c ZebedeeClient) getBreadcrumb(uri string) ([]model.TaxonomyNode, error) {
+	b, err := c.get("/parents?uri=" + uri)
+	if err != nil {
+		return nil, err
+	}
+
+	var parentsJSON []models.TaxonomyNode
+	if err = json.Unmarshal(b, &parentsJSON); err != nil {
+		return nil, err
+	}
+
+	var parents []model.TaxonomyNode
+	for _, value := range parentsJSON {
+		parents = append(parents, model.TaxonomyNode{
+			Title: value.Description.Title,
+			URI:   value.URI,
+		})
+	}
+
+	return parents, nil
+}
+
+func (c ZebedeeClient) populateStaticLandingPageModel(dlp models.DatasetLandingPage) (StaticDatasetLandingPage, error) {
+	//Map Zebedee response data to new page model
+	var sdlp StaticDatasetLandingPage
+	sdlp.Type = dlp.Type
+	sdlp.URI = dlp.URI
+	sdlp.Metadata.Title = dlp.Description.Title
+	sdlp.Metadata.Description = dlp.Description.Summary
+	sdlp.DatasetLandingPage.Related.Datasets = dlp.RelatedDatasets
+	sdlp.DatasetLandingPage.Related.Publications = dlp.RelatedDocuments
+	sdlp.DatasetLandingPage.Related.Methodology = append(dlp.RelatedMethodology, dlp.RelatedMethodologyArticle...)
+	sdlp.DatasetLandingPage.Related.Links = dlp.RelatedLinks
+	sdlp.DatasetLandingPage.IsNationalStatistic = dlp.Description.NationalStatistic
+	sdlp.DatasetLandingPage.IsTimeseries = dlp.Timeseries
+	sdlp.ContactDetails.Email = dlp.Description.Contact.Email
+	sdlp.ContactDetails.Telephone = dlp.Description.Contact.Telephone
+	sdlp.ContactDetails.Name = dlp.Description.Contact.Name
+	sdlp.DatasetLandingPage.ReleaseDate = dlp.Description.ReleaseDate
+	sdlp.DatasetLandingPage.NextRelease = dlp.Description.NextRelease
+	sdlp.DatasetLandingPage.Notes = dlp.Section.Markdown
+	sdlp.FilterID = dlp.FilterID
+	bc, err := c.getBreadcrumb(dlp.URI)
+	if err != nil {
+		return StaticDatasetLandingPage{}, err
```

Comment:
could consider just returning `sdlp` rather than throwing away existing data, since we've done the work to get it? and mention in the godoc that if an error is returned you may only get partial data?